File: backend/todos-service/src/main/java/com/ironhack/todosservice/model/Todo.java
package com.ironhack.todosservice.model;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import java.util.Date;

@Entity
public class Todo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private Date creationDate;
    private String description;
    private Date dueDate;
    private boolean isDone;

    public Todo(Date creationDate, String description, Date dueDate, boolean isDone) {
        this.creationDate = new Date();
        this.description = description;
        this.dueDate = dueDate;
        this.isDone = isDone;
    }

    public Todo(Integer id, Date creationDate, String description, Date dueDate, boolean isDone) {
        this.id = id;
        this.creationDate = new Date();
        this.description = description;
        this.dueDate = dueDate;
        this.isDone = isDone;
    }

    public Todo() {
        this.creationDate = new Date();
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Date getCreationDate() {
        return creationDate;
    }

    public void setCreationDate(Date creationDate) {
        this.creationDate = creationDate;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDueDate() {
        return dueDate;
    }

    public void setDueDate(Date dueDate) {
        this.dueDate = dueDate;
    }

    public boolean isDone() {
        return isDone;
    }

    public void setDone(boolean done) {
        isDone = done;
    }
}

File: frontend/src/app/model/todo.ts
export class Todo {

    private _creationDate: Date;
    private _isDone: boolean;


    constructor(
        private _id: number, 
        private _description: string, 
        private _dueDate?: Date
    ){}
    
    public get isDone(): boolean {
        return this._isDone;
    }
    public set isDone(value: boolean) {
        this._isDone = value;
    }
    public get dueDate(): Date {
        return this._dueDate;
    }
    public set dueDate(value: Date) {
        this._dueDate = value;
    }
    public get description(): string {
        return this._description;
    }
    public set description(value: string) {
        this._description = value;
    }
    public get creationDate(): Date {
        return this._creationDate;
    }
    public set creationDate(value: Date) {
        this._creationDate = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

}
